### PR TITLE
ESMF@8.5.0, MAPL@2.40.3 as defaults

### LINKED ISSUE #753: [INSTALL] Add mapl@2.40.3 or mapl@2.35.2 with esmf@8.5.0 to spack-stack@1.5.0 environments
**Which software in the stack would you like installed?**
Per https://github.com/ufs-community/ufs-weather-model/issues/1854, we need to add `mapl@2.40.3` *or* `mapl@2.35.2` (depends on testing that is currently going on) with `esmf@8.5.0` to the `spack-stack@1.5.0` environments.

**What is the version/tag of the software?**
See above

**What compilation options would you like set?**
Same as before

**Installation timeframe: Would you like this package to be installed in an upcoming quarterly spack-stack release, or sooner? If it is needed sooner (such as for testing a new package/version before incorporating it into a full release), please specify a preferred timeline and *one* machine on which you would like it to be installed for testing prior to the next spack-stack release.**
Roll out as part of `spack-stack@1.5.0`

**Any other relevant information that we should know to correctly install the software??**
n/s

**Additional context**
A test environment will be installed on Hera based on the current ufs-weather-model spack-stack environment to enable testing before the `spack-stack@1.5.0` release.

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

Update ESMF and MAPL defaults per Oct 19 spack-stack meeting.

### Testing

UFS WM cpld_control_p8/intel successful on Orion; cpld_control_gfsv17 successful on Hera

### Applications affected

all

### Systems affected

all

### Dependencies

https://github.com/JCSDA/spack/pull/352

### Issue(s) addressed

Fixes #753

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji this should be ready to go very shortly. Oddly, in just the ubuntu CI it was failing because of duplicate py-pandas (mix of 1.5.2 and 1.5.3). Do you have any concerns about setting the default to 1.5.2? I know you had some issues once upon a time with that setting.

**Comment by climbfuji** [comment]
> @climbfuji this should be ready to go very shortly. Oddly, in just the ubuntu CI it was failing because of duplicate py-pandas (mix of 1.5.2 and 1.5.3). Do you have any concerns about setting the default to 1.5.2? I know you had some issues once upon a time with that setting.

We need to verify that the excel dependencies are getting installed correctly. Do we know why it insists on 1.5.2?

**Comment by AlexanderRichert-NOAA** [comment]
No. For some reason, only `ufs-weather-model-env>ufs-pyenv>py-pandas` was using 1.5.2, everything else was using 1.5.3. And it's only in the ubuntu CI (not macOS and not on Hera).

**Comment by climbfuji** [comment]
> No. For some reason, only `ufs-weather-model-env>ufs-pyenv>py-pandas` was using 1.5.2, everything else was using 1.5.3. And it's only in the ubuntu CI (not macOS and not on Hera).

Can we force it to use 1.5.3 since that's the version used everywhere else?

**Comment by AlexanderRichert-NOAA** [comment]
Yep, looks like that works okay.

**Comment by climbfuji** [comment]
> Yep, looks like that works okay.

Great, thanks!

File: configs/templates/unified-dev/spack.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ spack:
       # Various fms tags (list all to avoid duplicate packages)
       - fms@release-jcsda
       - fms@2023.01
```
Comment (climbfuji) [inline configs/templates/unified-dev/spack.yaml:30]: Did we want to remove `fms@2023.01`?

Comment (AlexanderRichert-NOAA) [reply configs/templates/unified-dev/spack.yaml:30]: I was going to but then noticed it's hardcoded into ufs-srw-app and one other metapackage, so for now I'd prefer to not go down that rabbit hole unless necessary.

Comment (climbfuji) [reply configs/templates/unified-dev/spack.yaml:30]: Make sense!

**Review by climbfuji — COMMENTED** [review]
Looks good to me. I merged the spack PR for fms and gftl-shared, please update the submodule pointer. Thanks!

**Comment by AlexanderRichert-NOAA** [comment]
Done, thanks!

**Review by climbfuji — APPROVED** [review]

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA Why did you close this PR instead of merging it? Can I restore your branch and reopen the PR?

**Comment by climbfuji** [comment]
Actually, I am going to merge it since I already approved it and it passed CI.

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji thanks for taking care of this, I must have absentmindedly hit the close button :upside_down_face:

**Pull request merged**